shell/common/shell.cc

```diff
@@ -1401,10 +1401,20 @@ bool Shell::OnServiceProtocolRunInView(
   configuration.SetEntrypointAndLibrary(engine_->GetLastEntrypoint(),
                                         engine_->GetLastEntrypointLibrary());
 
-  configuration.AddAssetResolver(
-      std::make_unique<DirectoryAssetBundle>(fml::OpenDirectory(
-          asset_directory_path.c_str(), false, fml::FilePermission::kRead)));
-  configuration.AddAssetResolver(RestoreOriginalAssetResolver());
+  configuration.AddAssetResolver(std::make_unique<DirectoryAssetBundle>(
+      fml::OpenDirectory(asset_directory_path.c_str(), false,
+                         fml::FilePermission::kRead),
+      false));
+
+  // Preserve any original asset resolvers to avoid syncing unchanged assets
+  // over the DevFS connection.
+  auto old_resolvers = engine_->GetAssetManager()->TakeResolvers();
+  for (uint64_t i = 0; i < old_resolvers.size(); i++) {
```

Comment:
use something like `for (auto& old_resolver : engine_->GetAssetManager()->TakeResolvers())`

Reply:
I think I need to handle asset manager being a nullptr too, right?

Reply:
yes - looking at the `Engine` it appears that the asset manager could be null if something went wrong during initialization or a restart